generateBuildrootSBOM.py

```diff
@@ -44,23 +44,24 @@ def create_buildroot_sbom(args, br_bom):
     # the resulting JSON. Use a try/except block to help with run time issues.
     with open(args.input_file, newline='') as csvfile:
         sheetX = csv.DictReader(csvfile)
-        my_component_list = []
+
         for row in sheetX:
             try:
-                purl_info: str | Any = "pkg:generic/" + row['PACKAGE'] + "@" + row['VERSION'] + \
-                                       "?download_url=" + row['SOURCE SITE'] + row['SOURCE ARCHIVE']
+                # purl_info: str | Any = "pkg:generic/" + row['PACKAGE'] + "@" + row['VERSION'] + \
+                #                       "?download_url=" + row['SOURCE SITE'] + row['SOURCE ARCHIVE']
 
                 from packageurl import PackageURL
 
+                purl_info = PackageURL(type='generic', name=row['PACKAGE'], version=row['VERSION'],
+                                       qualifiers={'download_url': row['SOURCE SITE'] + row['SOURCE ARCHIVE']})
+
                 componenttype = cyclonedx.model.component.ComponentType('firmware')
```

Comment:
you are supposed to use the constant, not the value.

```patch
- componenttype = cyclonedx.model.component.ComponentType('firmware')
+ componenttype = cyclonedx.model.component.ComponentType.FIRMWARE 
````

Reply:
I have a serious misunderstanding of instantiating the object based upon looking at the file .../cyclonedx/model/component.py 
class ComponentType(str, Enum):
   from my years of programing experience my natural expectation is to use the Enum argument for my call. I was fooled in this case. I am a C language programmer for decades, with some experience with C++. This Enum construct using the dot notation to set a value baffles me. Honestly I would have no idea how to do this assignment.